### DISCUSSION TITLE
Github actions for Ethereum or Blockchain

### DISCUSSION THREAD

**Discussion by jveer634** [EddieHubCommunity/support#3696] [Q&A]
Hello People,
As I was thinking I would like to use the power of Github Actions, so does anyone know any useful actions particularly designed for blockchain except for linting and testing.

**Comment by asciinish** [comment]
@jveer634 , Sincerely speaking, there is not a lot you can do to **Ethereum** or **Blockchain,** except for _linting_ and _testing_. Nevertheless, here are some other important pipelines, which can be automated using Github Actions.

1. Continuous Deployment: Automated deployment of your smart contracts to new test networks like Ropsten or Rinkeby. You may also deploy these contracts into the Mainnet, whenever changes are made into specific branches. Furthermore, you may also use the same actions to conjure frontend web-apps that interact smartly with your Ethereum and Blockchain network.

2. Security Analysis: If not linting or testing, maybe security analysis can be done through Github actions. You may use Slither or MythX to check your code and analyse any potential vulnerabilities. 

3. Monitoring : Set up GitHub Actions to monitor your Ethereum nodes or Blockchain networks for specific events or conditions and trigger alerts or notifications accordingly. This could involve checking for contract interactions, monitoring network health, or tracking gas prices.

Obviously, there are other parameters, which can be managed using Github actions, but I have listed the technical ones only. Furthermore, you may use actions to automate any document generation to the specific repository, create an automated protocol for release and version management, automate the workflow(scheduling updates, node disintegration, or market-surveys).

Regards
